modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperCppEngine.java

```diff
@@ -0,0 +1,417 @@
+      command.add(String.format("%f", whispercppLogProbThreshold.get()));
+    }
+    if (whispercppDiarization.isSome() && whispercppDiarization.get()) {
+      command.add("-di");
+    }
+    if (whispercppTinyDiarization.isSome() && whispercppTinyDiarization.get()) {
+      command.add("-tdrz");
+    }
+    if (whispercppNoFallback.isSome() && whispercppNoFallback.get()) {
+      command.add("-nf");
+    }
+
+    String subtitleLanguage;
+
+    // set language of the source audio if known
+    if (!language.isBlank()) {
+      logger.info("Using language {} from workflows", language);
+      command.add("--language");
+      command.add(language);
+    } else {
+      logger.debug("Auto-detecting language");
+      command.add("--language");
+      command.add("auto");
+    }
+
+    if (translate) {
+      command.add("--translate");
+      logger.info("Translation enabled");
+      subtitleLanguage = "en";
+    } else {
+      subtitleLanguage = language;
+    }
+
+
+    logger.info("Executing WhisperC++'s transcription command: {}", command);
+
+    Process process = null;
+
+    try {
+      ProcessBuilder processBuilder = new ProcessBuilder(command);
+      processBuilder.redirectErrorStream(true);
+      processBuilder.redirectInput(ProcessBuilder.Redirect.PIPE)
+          .redirectError(ProcessBuilder.Redirect.PIPE)
+          .redirectOutput(ProcessBuilder.Redirect.PIPE);
+      process = processBuilder.start();
+
+      try (BufferedReader in = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+        String line;
+        while ((line = in.readLine()) != null) { // consume process output
+          logger.debug(line);
+        }
+      }
+
+      // wait until the task is finished
+      int exitCode = process.waitFor();
+      logger.info("WhisperC++ process finished with exit code {}",exitCode);
+
+      if (exitCode != 0) {
+        var error = "";
+        try (var errorStream = process.getInputStream()) {
+          error = "\n Output:\n" + IOUtils.toString(errorStream, StandardCharsets.UTF_8);
+        }
+        throw new SpeechToTextEngineException(
+            String.format("WhisperC++ exited abnormally with status %d (command: %s)%s", exitCode, command, error));
+      }
+
+      if (!preparedOutputFile.isFile()) {
+        throw new SpeechToTextEngineException("WhisperC++ produced no output");
+      }
+      logger.info("Subtitles file generated successfully: {}", preparedOutputFile);
+    } catch (Exception e) {
+      logger.info("Transcription failed closing WhisperC++ transcription process for: {}", mediaFile);
+      throw new SpeechToTextEngineException(e);
+    } finally {
+      IoSupport.closeQuietly(process);
+    }
+
+    // Detect language if not set
+    if (subtitleLanguage.isBlank()) {
+      JSONParser jsonParser = new JSONParser();
+      try {
+        FileReader reader = new FileReader(preparedOutputFile.getAbsolutePath().replaceFirst("[.][^.]+$", "")
+            + ".json");
```

Comment:
I'm writing this down here since this operated on the JSON file, although this is a more general problem. This code will cause Whisper to generate a JSON file which is never cleaned up. The changes made necessary by pull request #5472 will also fix this and you wouldn't need to clean up every file, but waiting for that PR also means some minor changes to this methods interface.

Reply:
I added the cleanup, so this PR can be merged anytime before or after #5472